pkg/daemon/daemon.go

```diff
@@ -361,23 +329,32 @@ func (dn *Daemon) BindPodMounts() error {
 }
 
 func (dn *Daemon) runLoginMonitor(stopCh <-chan struct{}, exitCh chan<- error) {
-	sessionNewCh := dn.loginClient.Subscribe("SessionNew")
-	for {
-		select {
-		case <-stopCh:
-			return
-		case msg, ok := <-sessionNewCh:
-			if !ok {
-				glog.V(4).Info("Not adding the ssh accessed annotation because the logind SessionNew channel is closed")
+	go func() {
+		cmd := exec.Command("journalctl", "-b", "-f", "-o", "cat", "MESSAGE_ID=" + sdMessageSessionStart)
+		stdout, err := cmd.StdoutPipe()
+		if err != nil {
+			glog.Fatal(err)
+		}
+		if err := cmd.Start(); err != nil {
+			glog.Fatal(err)
+		}
+		buf := make([]byte, 1024)
+		l, err := stdout.Read(buf)
+		if err != nil {
+			if err == io.EOF {
 				return
 			}
-			glog.Infof("Detected a new login session: %v", msg)
+			glog.Fatal(err)
```

Comment:
all these `err` should be sent to the `exitCh` for correct reporting and stop. These should follow with a `return` to exit the goroutine and cleanup now that we don't listen on `stopCh` anymore.

On top of that, `stopCh` must still be around to signal this goroutine to exit

Reply:
OK, fixed.